ios/l2_interfaces/ios_l2_interface.yaml

```diff
@@ -0,0 +1,74 @@
+---
+GENERATOR_VERSION: '1.0'
+
+ANSIBLE_METADATA: |
+    {
+      'metadata_version': '1.1',
+      'status': ['preview'],
+      'supported_by': 'network'
+    }
+NETWORK_OS: ios
+RESOURCE: l2_interfaces
+COPYRIGHT: Copyright 2019 Red Hat
+
+DOCUMENTATION: |
+  module: ios_l2_interfaces
+  version_added: 2.9
+  short_description: Manage Layer-2 interface on Cisco IOS devices.
+  description: This module provides declarative management of Layer-2 interface on Cisco IOS devices.
+  author: Sumit Jaiswal (@justjais)
+  options:
+    config:
+      description: A dictionary of Layer-2 interface options
+      type: list
+      elements: dict
+      suboptions:
+        name:
+          description:
+          - Full name of interface, e.g. ge-0/0/0.
+          type: str
+          required: True
+        mode:
+          default: access
+          description:
+          - Mode in which interface needs to be configured
+          type: str
+          choices: ['access', 'trunk']
+        access_vlan:
```

Comment:
If we add a root access key and root trunk key, we can get rid of mode and nest the trunk paramters under the trunk key.
```
access:
  vlan:
trunk:
  vlans: (comma delimited list or all)
  native_vlan:
```

We will need to do the analysis of what vlans will need to be added or removed by splitting the comma delimited list and comparing it to the config to determine the add or remove.

Reply:
`vlans` should be an actual list imho

Reply:
agreed! It should be a valid YAML list.

Reply:
done